src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -7845,14 +7847,17 @@ private GotoStatementSyntax ParseGotoStatement()
 
         private IfStatementSyntax ParseIfStatement()
         {
-            Debug.Assert(this.CurrentToken.Kind == SyntaxKind.IfKeyword);
+            Debug.Assert(this.CurrentToken.Kind == SyntaxKind.IfKeyword || this.CurrentToken.Kind == SyntaxKind.ElseKeyword);
+
             var @if = this.EatToken(SyntaxKind.IfKeyword);
             var openParen = this.EatToken(SyntaxKind.OpenParenToken);
             var condition = this.ParseExpressionCore();
             var closeParen = this.EatToken(SyntaxKind.CloseParenToken);
-            var statement = this.ParseEmbeddedStatement();
-            var elseClause = ParseElseClauseOpt();
-
+            // when 'if' is missing, then we parse an expression statement in a standard way (which adds missing syntax properly)
+            // parsing embedded one expects non-null statement to be present
+            var statement = this.CurrentToken.Kind == SyntaxKind.ElseKeyword ? this.ParseExpressionStatement() : this.ParseEmbeddedStatement();
```

Comment:
note: if you do want the current form, then the comment needs to be fixed up  Because it applies to *both* when an 'if' is missing, *and* when tehre is no statement following the close-paren.

Reply:
I followed your https://github.com/dotnet/roslyn/pull/28115#discussion_r198252144 but most likely missed the point in the comment. I'll change the code to the what you proposed last.